lib/luminork-server/src/search/parser.rs

```diff
@@ -0,0 +1,555 @@
+use super::{
+    Error,
+    Result,
+    SearchQuery,
+    SearchTerm,
+};
+
+/// Parses a search query.
```

Comment:
The parser is written to be a direct copy of the frontend `search.ts` parser. I think there are better ways to do this that will be more robust, but it seemed prudent to *start* by replicating the logic as closely as possible, to maximize the chance we've got it right.